src/Reflex/TriggerEvent/Base.hs

```diff
@@ -29,21 +33,28 @@ import Reflex.TriggerEvent.Class
 data TriggerInvocation a = TriggerInvocation a (IO ())
 
 -- | A reference to an 'EventTrigger' suitable for firing with 'TriggerEventT'.
-newtype EventTriggerRef t m a = EventTriggerRef { unEventTriggerRef :: Ref m (Maybe (EventTrigger t a)) }
+newtype EventTriggerRef t a = EventTriggerRef { unEventTriggerRef :: IORef (Maybe (EventTrigger t a)) }
```

Comment:
I don't think it'll be hard to change when that time comes. `IORef` is already hard coded in a number of places.